tensorboard/webapp/header/header.container.ts

```diff
@@ -15,17 +15,14 @@ limitations under the License.
 import {Component} from '@angular/core';
 import {Store, select, createSelector} from '@ngrx/store';
 
-import {State, getActivePlugin, getPlugins} from '../../core/core.reducers';
-import {changePlugin} from '../../core/core.actions';
+import {getActivePlugin, getPlugins, State} from '../core/store';
+import {changePlugin} from '../core/actions';
 
-import {PluginMetadata, PluginId} from '../../types/api';
+import {PluginId} from '../types/api';
+import {UiPluginMetadata} from './types';
 
 /** @typehack */ import * as _typeHackRxjs from 'rxjs';
 
-export interface UiPluginMetadata extends PluginMetadata {
-  id: PluginId;
-}
-
 const getUiPlugins = createSelector(
```

Comment:
For my understanding, just to confirm: Containers can define their own Selectors in *container.ts instead of from core/store/selectors.ts or <feature>/store/selectors.ts?  so long as the logic is specific to the Container?

Reply:
Correct.  In this context, 'selector' is just a synonym for 'memoized observable'.  Critically, it does not reach inside the NgRx state, because it obtains data from another Selector imported from `.../store` (`getPlugins`, in this case).

Reply:
Thanks for the explanation.  In the design doc, I didn't realize that selectors can be defined in both places.  It seems very reasonable though, since these selectors are deriving information that is close to where it is used.